qiskit_experiments/framework/composite/composite_experiment.py

```diff
@@ -39,10 +40,14 @@ def __init__(
             qubits: list of physical qubits for the experiment.
             backend: Optional, the backend to run the experiment on.
             experiment_type: Optional, composite experiment subclass name.
+            analysis: Optional, the composite analysis class to use. If not
+                      provided this will be initialized automatically from the
+                      supplied experiments.
         """
         self._experiments = experiments
         self._num_experiments = len(experiments)
-        analysis = CompositeAnalysis([exp.analysis for exp in self._experiments])
+        if analysis is None:
```

Comment:
We need instance check. If user provides custom analysis that directly inherits from the base analysis, the mechanism to instantiate child container is missing there.